packages/client/wallets/smart-wallet/src/SmartWalletSDK.ts

```diff
@@ -83,4 +92,23 @@ export class SmartWalletSDK {
             }
         });
     }
+
+    private assertValidChain(chain: SmartWalletChain) {
+        if (!this.validChain(chain)) {
+            const validChains = this.crossmintEnv === "production" ? SMART_WALLET_MAINNETS : SMART_WALLET_TESTNETS;
+            const formattedChains = validChains.map((chain) => `"${chain}"`).join(", ");
+            throw new SmartWalletError(
+                `Invalid chain "${chain}" for environment "${this.crossmintEnv}" specified by API key.\n` +
```

Comment:
`The selected chain "${chain}" is not available in "${this.crossmintEnv}". Either set a valid chain or check you're using an API key for the environment you're trying to target.`

Reply:
updated